### DISCUSSION TITLE
It can be displayed normally locally, but it is invalid to install it in the extension market after packaging. The code will run, but nothing is displayed under the registered treeview

### DISCUSSION THREAD

**Discussion by dddssw** [microsoft/vscode-discussions#1584] [Extension Development QnA]
local:
![image](https://github.com/user-attachments/assets/7549a644-c97a-45bf-a946-5432cd45d1ac)
online:
![image](https://github.com/user-attachments/assets/6ef80997-905f-45f7-9640-1e585b4af153)

I don't know how to debug

**Comment by gjsjohnmurray** [comment]
Any clues on the Extension Host channel of the Output tab on the Panel?

My guess is your VSIX doesn't contain all of the packages your extension depends on.

**Comment by dddssw** [comment]
I use webpack for packaging, and it works fine locally. These packages should all be packaged in.

**Reply by gjsjohnmurray** [reply]
There's a command line option for `vsce` that shows you what it is packaging. I suggest you check this.

**Comment by dddssw** [comment]
I'm getting an error when using pnpm to package, but it works fine when using npm to download dependencies. I don't know if it's related to this.

**Comment by alefragnani** [comment]
Hi @dddssw ,

When you say _local_, I suppose you are referring to _while debugging_ right? 

If that’s the case, don’t use debug. Instead, package the extension using `vsce package` to create the `.vsix` package and install it in VS Code using the `Install from vsix` command. This would expose any “dependency issue” you may have on your extension. 

If you don’t have any dependency issue, maybe your “scenario” is not exactly the same on both tests. Since you didn’t provide any details on what/how the treeview is filled, it’s hard to suppose what is wrong. Are you opening single root folders, has any folder opened at all, is selecting the same kind of file? If both (local and online) scenarios are exactly the same, you should look for erros in the `Console` view in the `Toggle Developer Tools` command.

Hope this helps

**Reply by dddssw** [reply]
My tree data is obtained like this

```
 async getImportData(files: string[], hooksPath: string) {
    const hookPromises = files.map(async (pathItem) => {
      if (pathItem.slice(-2) === "ts") {
        return;
      }
      const pathfile = path.join(hooksPath, pathItem);
      const fileURL = pathToFileURL(pathfile).toString();
      const jsPath = fileURL.slice(0, -2) + "js";
      const filePathJs = path.join(hooksPath, pathItem).slice(0, -2) + "js";
      console.log(filePathJs, "filePathJs");
      const data = fs.readFileSync(filePathJs, "utf8");
      const updatedData = data.replace(/^import.*;$/gm, "");
      // 同步写入修改后的内容
      fs.writeFileSync(filePathJs, updatedData, "utf8");

        const hook = await import(/* webpackIgnore: true */ jsPath);
        const returnsObj: any = {};
        const paramsObj: any = {};
        for (const key in hook) {
          const name = key === "default" ? pathItem.slice(0, -3) : key;
          const returns = findReturn(filePathJs, name);
          console.log(returns, filePathJs, name, "returns");
          const params = findParams(filePathJs, name);
          returnsObj[key] = returns;
          paramsObj[key] = params;
          console.log(returns, params, "sss1");
        }
        let isExport = 1;
        if (!hook) {
          isExport = 0;
        }
        return new Hooks(
          pathItem.slice(0, -3),
          isExport,
          hook,
          "",
          pathItem.slice(0, -3),
          "",
          returnsObj,
          paramsObj
        );
    });

    // 等待所有异步操作完成
    const hookList = await Promise.all(hookPromises);

    console.log(hookList, "hooklist1");
    return hookList;
  }
```
I suspect there's a problem here.
```
  const hook = await import(/* webpackIgnore: true */ jsPath);
```
And there is one more error than locally. I guess vscode does not support this kind of esm，But I don't know how to solve it
![image](https://github.com/user-attachments/assets/593d1472-1b14-42d1-92a8-aa12c4a3cd69)
